libshaderc_util/src/compiler.cc

```diff
@@ -67,40 +67,61 @@ std::pair<int, string_piece> DecodeLineDirective(string_piece directive) {
   return std::make_pair(line, directive);
 }
 
-// Writes the message contained in the diagnostic parameter to *dest. Assumes
-// the diagnostic message is reported for a binary location.
-void OutputSpvToolsDiagnostic(spv_diagnostic diagnostic, std::string* dest) {
-  assert(!diagnostic->isTextSource);
-
-  std::ostringstream os;
-  os << diagnostic->position.index << ": " << diagnostic->error;
-  *dest = os.str();
-}
+// A wrapper class for SPIRV-Tools API. Uses singleton design pattern here to
+// amortize the cost of creating the SPIRV-Tools context object.
+class SpvtoolsApi {
+ public:
+  ~SpvtoolsApi() { spvContextDestroy(context_); }
+
+  // Returns the single API instance. The compiler guarantees that this instance
+  // is lazy initialized and will be destructed at program termination. The
+  // initialization is thread-safe under C++11 semantics.
+  static SpvtoolsApi& GetInstance() {
+    static SpvtoolsApi instance;
```

Comment:
I'm afraid this violates the style guide for the same reasons discussed here:

https://github.com/KhronosGroup/SPIRV-Tools/issues/53